lib/src/cancelable_operation.dart

```diff
@@ -174,8 +178,11 @@ class CancelableCompleter<T> {
 
   /// Completes [operation] to [value].
   ///
-  /// If [value] is a [Future], this will complete to the result of that
-  /// [Future] once it completes.
+  /// If [value] is a Future the [operation] will complete with the result of
+  /// that Future once it is available.
+  /// In that case [isComplete] will be true before the [operation] is complete.
+  ///
+  /// This method may not be called if [isComplete] is true.
```

Comment:
Mention that you must only omit the parameter or pass `null` if `T` is nullable.
(That's the requirement inherited from `Completer.complete`. It was a mistake to make that argument optional, but it made sense when everything was nullable).